numpy/core/_methods.py

```diff
@@ -56,14 +56,25 @@ def _any(a, axis=None, dtype=None, out=None, keepdims=False):
 def _all(a, axis=None, dtype=None, out=None, keepdims=False):
     return umr_all(a, axis, dtype, out, keepdims)
 
-def _count_reduce_items(arr, axis):
-    if axis is None:
-        axis = tuple(range(arr.ndim))
-    if not isinstance(axis, tuple):
-        axis = (axis,)
-    items = 1
-    for ax in axis:
-        items *= arr.shape[mu.normalize_axis_index(ax, arr.ndim)]
+def _count_reduce_items(arr, axis, keepdims=False, where=True):
+    if isinstance(where, bool):
```

Comment:
We should check which cases `ufunc.reduce` (c function name is something like GenericReduction) has this optimization for, and try the same ones. In particular, I'm trying to remember if we optimized for `np.bool_`.

Reply:
I found the `PyArray_GenericReduceFunction` in `numpy/core/src/multiarray/number.c` however that function looks more like a wrapper, being only 15 lines long. Could you point me in the right direction?

Reply:
I think @eric-wieser is indicating at this function : PyUFunc_GenericReduction in numpy/core/src/umath/ufunc_object.c . I wasn't able to find any special casing for np.bool except that where = True is being treated as if no where is provided.

Reply:
OK thank you for checking @anirudh2290.